### PR TITLE
Rework transformer so that it walks the schema only on initialization

### PR DESCRIPTION
**Pull request by tkuhlman**

### CODE REVIEW

File: transform/transform.go [hunk 1]
```diff
@@ -185,23 +185,14 @@ func (tis *transformInstructions) transform(in interface{}, fieldType string) (i
...
-		for _, instruction := range instructions.From {
-			newPath := strings.Replace(instruction.JsonPath, old, new, 1)
-			newInstructions.From = append(newInstructions.From,
-				transformInstruction{
-					JsonPath:   newPath,
-					Operations: instruction.Operations,
-				})
+// replaceJSONPathPrefix will switch old for new in the path of the transform instructions if the path starts with
+// old.
+func (tis *transformInstructions) replaceJSONPathPrefix(old, new string) {
```
Comment (ahornerr) [inline transform/transform.go]: 👍

File: transform/transformer.go [hunk 2]
```diff
@@ -20,44 +19,58 @@ import (
 // More details on the transform section of the schema are found at
 // https://github.com/GannettDigital/jstransform/blob/master/transform.adoc
 type Transformer struct {
-	in                  interface{}
-	relativePath        string
 	schema              *jsonschema.Schema
-	skipPrefix          string
 	transformIdentifier string // Used to select the proper transform Instructions
-	transformed         map[string]interface{}
+	root                instanceTransformer
```
Comment (ahornerr) [inline transform/transformer.go]: This is much cleaner

File: transform/instance.go [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+	}
+	return newArray, nil
+}
+
+// objectTransformer represents a JSON instance of type object and associated transforms.
+type objectTransformer struct {
+	children     map[string]instanceTransformer
+	defaultValue map[string]interface{}
+	jsonPath     string
+	transforms   *transformInstructions
```
[the remark this answers is not in the document]
Comment (tkuhlman) [reply transform/instance.go]: I totally agree but couldn't come up with a better name.

File: transform/instance.go [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+}
+
+func newScalarTransformer(path, transformIdentifier string, raw json.RawMessage, instanceType string) (*scalarTransformer, error) {
+	st := &scalarTransformer{
+		jsonType: instanceType,
+		jsonPath: path,
+	}
+
+	if instanceType == "string" {
+		instanctFormat, err := jsonparser.GetString(raw, "format")
```
[the remark this answers is not in the document]
Comment (tkuhlman) [reply transform/instance.go]: whoops I fix that up

**Pull request merged**